File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/PlacedAheadFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

import java.time.Instant;
import java.util.Optional;

@Component
public class PlacedAheadFunction implements AheadTimeFunction {

    @Override
    public Optional<Instant> getRelatedTime(SettledBet bet) {
        return Optional.ofNullable(bet.getPlacedOrActionDate().toInstant());
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/PlacedAheadFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.RealizedBet
import org.springframework.stereotype.Component
import java.time.Instant

@Component
class PlacedAheadFunction : AheadTimeFunction {

    override fun getRelatedTime(bet: RealizedBet): Instant? {
        return bet.settledBet.placed
    }

}
